var/spack/repos/builtin/packages/py-urllib3/package.py

```diff
@@ -23,22 +25,27 @@ class PyUrllib3(PythonPackage):
     version("1.20", sha256="97ef2b6e2878d84c0126b9f4e608e37a951ca7848e4855a7f7f4437d5c34a72f")
     version("1.14", sha256="dd4fb13a4ce50b18338c7e4d665b21fd38632c5d4b1d9f1a1379276bd3c08d37")
 
-    variant("socks", default=False, description="SOCKS and HTTP proxy support")
-    variant("secure", default=False, description="Add SSL/TLS support")
     variant("brotli", default=False, description="Add Brotli support")
+    variant("secure", default=False, description="Add SSL/TLS support")
+    variant("socks", default=False, description="SOCKS and HTTP proxy support")
+
+    depends_on("py-hatchling@1.6:1", when="@2:", type="build")
 
-    depends_on("python@2.7:2.8,3.4:", when="@:1.25", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@1.26.6", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.6:3", when="@1.26.12:", type=("build", "run"))
+    depends_on("brotli@1.0.9:", when="+brotli @2:", type=("build", "run"))
```

Comment:
Should we just group these under a `with when("+brotli"):`? I know py-brotlipy is no longer a dep but it feels weird to separate them. Do we need to add a py-brotli package? Also see https://github.com/spack/spack/pull/38823#discussion_r1264562482 if you want to get technical.

Reply:
I have added the `py-brotli` package as suggested.